src/model/gateway.rs

```diff
@@ -425,10 +444,23 @@ impl<'de> Deserialize<'de> for Activity {
             #[cfg(feature = "unstable_discord_api")]
             session_id,
             url,
+            buttons,
         })
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[non_exhaustive]
```

Comment:
On the Discord, someone noted that AcitivyButton is not constructable due to this `#[non_exhaustive]` ((link)[https://discord.com/channels/381880193251409931/919310428344029265/1170727369531727993]). Was this an oversight, or is there another way to construct ActivityButton?